File: ruby/wc.rb
def wc( fname, fhandle )
	lines = bytes = words = 0
	fhandle.each { |l|
		bytes += l.length
		words += l.chomp.split(/\s+/).reject { |w| w.length == 0 }.length
		lines += 1
	}
	print "%s\t%d\t%d\t%d\n" % [ fname, lines, words, bytes ]
end

if ( ARGV.length == 0 )
	wc( '', STDIN )
else
	ARGV.each { |fn| wc( fn, File.open( fn ) ) }
end

File: js/lib/wc.js
var util = require('util');

module.exports = function ( fname, fhandle, completed ) {
	var lines = 0;
	var words = 0;
	var bytes = 0;
	var lastws = false;
	fhandle.on('data',function(buf) {
		buf.toString().split('').forEach(function(val, index, array) {
			if ( val == "\n" ) lines += 1;
			if ( val == " " || val == "\t" ) {
				if ( lastws == false ) words += 1;
				lastws = true;
			} else lastws = false;
			bytes += 1;
		} );
	} );
	fhandle.on('end',function() {
		console.log( util.format( "%s\t%d\t%d\t%d", fname, lines, words, bytes ) );
		completed();
	});
}
